### PR TITLE
Safe cache write and validation of data read

### PR DESCRIPTION
**Pull request by texnikru**
Safe cache write and validation of data read. Only for 5.0 due 4.2 is only for security fixes.

### CODE REVIEW

File: src/Illuminate/Support/helpers.php [hunk 1]
```diff
@@ -784,3 +784,41 @@ function with($object)
...
+
+if ( ! function_exists('is_serialized'))
+{
+	/**
+	 * Check value to find if it was serialized.
+	 *
+	 * @param  mixed  $value
+	 * @return bool
+	 */
+
```
Comment (taiyaei) [inline src/Illuminate/Support/helpers.php]: No newline here

File: src/Illuminate/Support/helpers.php [hunk 2]
```diff
@@ -784,3 +784,41 @@ function with($object)
...
+	function is_serialized($value)
+	{
+		if ( ! is_string( $value ))
```
Comment (taiyaei) [inline src/Illuminate/Support/helpers.php]: No spaces surrounding argument lists

File: src/Illuminate/Support/helpers.php [hunk 3]
```diff
@@ -784,3 +784,41 @@ function with($object)
...
+		{
+			return false;
+		}
+ 		if ('N;' == $value)
+		{
+			return true;
+		}
+		if (strlen( $value ) < 4)
```
Comment (taiyaei) [inline src/Illuminate/Support/helpers.php]: same

File: src/Illuminate/Support/helpers.php [hunk 4]
```diff
@@ -784,3 +784,41 @@ function with($object)
...
+		{
+			return false;
+		}
+		if (':' !== $value[1])
+		{
+			return false;
+		}
+
+		$lastc = substr($value, -1);
+		if (';' !== $lastc && '}' !== $lastc)
```
Comment (taiyaei) [inline src/Illuminate/Support/helpers.php]: Also no yoda comparisons please, same for the `if`s above

**Comment by texnikru** [comment]
Done

File: src/Illuminate/Cache/FileStore.php [hunk 5]
```diff
@@ -76,7 +78,16 @@ protected function getPayload($key)
 			return array('data' => null, 'time' => null);
 		}
 
-		$data = unserialize(substr($contents, 10));
+		// Stored cache may be corrupted
+		$contents = substr($contents, 10);
+		if ( ! is_serialized($contents))
```
Comment (GrahamCampbell) [inline src/Illuminate/Cache/FileStore.php]: add a new line before this logical block please

File: src/Illuminate/Support/helpers.php [hunk 6]
```diff
@@ -784,3 +784,40 @@ function with($object)
...
+	 * @param  mixed  $value
+	 * @return bool
+	 */
+	function is_serialized($value)
+	{
+		if ( ! is_string($value))
+		{
+			return false;
+		}
+ 		if ($value == 'N;')
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/helpers.php]: there must be a blank line after each closing brace here

File: src/Illuminate/Support/helpers.php [hunk 7]
```diff
@@ -784,3 +784,40 @@ function with($object)
...
+		{
+			return false;
+		}
+		if ($value[1] !== ':')
+		{
+			return false;
+		}
+
+		$lastc = substr($value, -1);
+		if ($lastc !== ';' && $lastc !== '}')
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/helpers.php]: and one before this

**Comment by GrahamCampbell** [comment]
Please fix the remaining issues and squash to one commit.

**Comment by texnikru** [comment]
Closed for #8293

**Pull request closed without merging**